### PR TITLE
Add SelectedString property to DSDropDownBase class

### PR DESCRIPTION
**Pull request by ZiyunShang**
Please Note:
1. Before submitting the PR, please review [How to Contribute to Dynamo](https://github.com/DynamoDS/Dynamo/blob/master/CONTRIBUTING.md)
2. Dynamo Team will meet 1x a month to review PRs found on Github (Issues will be handled separately)
3. PRs will be reviewed from oldest to newest
4. If a reviewed PR requires changes by the owner, the owner of the PR has 30 days to respond. If the PR has seen no activity by the next session, it will be either closed by the team or depending on its utility will be taken over by someone on the team
5. PRs should use either Dynamo's default PR template or [one of these other template options](https://github.com/DynamoDS/Dynamo/wiki/Choosing-a-Pull-Request-Template) in order to be considered for review.
6. PRs that do not have one of the Dynamo PR templates completely filled out with all declarations satisfied will not be reviewed by the Dynamo team.
7. PRs made to the `DynamoRevit` repo will need to be cherry-picked into all the DynamoRevit Release branches that Dynamo supports. Contributors will be responsible for cherry-picking their reviewed commits to the other branches after a `LGTM` label is added to the PR.

### Purpose

Due to Categories Dropdown Produces Inconsistent Results Between Revit versions. Add the new property to check the right selected index.
Refer to https://github.com/DynamoDS/DynamoRevit/issues/2281

### Declarations

Check these if you believe they are true

- [ ] The code base is in a better state after this PR
- [ ] Is documented according to the [standards](https://github.com/DynamoDS/Dynamo/wiki/Coding-Standards)
- [ ] The level of testing this PR includes is appropriate
- [ ] User facing strings, if any, are extracted into `*.resx` files
- [ ] All tests pass using the self-service CI.
- [ ] Snapshot of UI changes, if any.
- [ ] Changes to the API follow [Semantic Versioning](https://github.com/DynamoDS/Dynamo/wiki/Dynamo-Versions), and are documented in the [API Changes](https://github.com/DynamoDS/Dynamo/wiki/API-Changes) document.

### Reviewers

@AndyDu1985 @QilongTang @mjkkirschner 

### FYIs

(FILL ME IN, Optional) Names of anyone else you wish to be notified of

### CODE REVIEW

File: src/Libraries/CoreNodeModels/DropDown.cs [hunk 1]
```diff
@@ -89,6 +89,27 @@ public int SelectedIndex
...
+        {
+            get { return selectedString; }
+            set
+            {
+                if(string.IsNullOrEmpty(selectedString))
+                {
+                    if(Items.Count()>0 && !string.IsNullOrEmpty(value))
+                    {
+                        var item = Items.FirstOrDefault(i => (i.Item).ToString().Equals(value));
+                        SelectedIndex = item != null ?
```
Comment (ZiyunShang) [inline src/Libraries/CoreNodeModels/DropDown.cs]: This reset SelectedIndex action is not quite fit the set method's function, but I can't find any other place to do this action. Do you have any other suggestions?

Comment (ZiyunShang) [reply src/Libraries/CoreNodeModels/DropDown.cs]: Hi @QilongTang , I have debugged in dropdown class, and there is a JsonConstructor now at https://github.com/DynamoDS/Dynamo/blob/0b0d90ba1bf6cad239b7f44522d981981fbd5511/src/Libraries/CoreNodeModels/DropDown.cs#L99 , and Dynamo DropDown Items would be set here. And then, SelectedIndex and SelectedString was set value later. If I want to get the matched SelectedIndex and SelectedString, the values of Index, String and Items is necessary.

**Comment by mjkkirschner** [comment]
Deserializecore is used for opening 1.3x xml graphs and copy paste.

**Comment by ZiyunShang** [comment]
@mjkkirschner Yes, and after version 2.0.0, SerializeCore and DeserializeCore were set Obsolete in ModelBase class at https://github.com/DynamoDS/Dynamo/blob/0622384b0bbffd5a2de617b8d4a8ad3aaeb72396/src/DynamoCore/Graph/ModelBase.cs#L361

**Comment by QilongTang** [comment]
@ZiyunShang I am going to open a PR soon for your reference, currently fixing things locally from my testing

**Pull request closed without merging**